inst/scripts/2.4.0/make-hermann-spermatogenesis-data.Rmd

```diff
@@ -0,0 +1,205 @@
+knitr::opts_chunk$set(error=FALSE, message=FALSE, warning=FALSE, eval=FALSE)
+```
+
+# Loading required R packages
+
+```{r}
+suppressPackageStartupMessages({
+    library(Biostrings)
+    library(BSgenome)
+    library(eisaR)
+    library(GenomicFeatures)
+    library(SummarizedExperiment)
+    library(tximeta)
+    library(rjson)
+    library(SingleCellExperiment)
+    library(dplyr)
+})
+```
+
+# Setting the working directory
+
+```{r}
+workdir <- tempdir()
+Sys.setenv(workdir = workdir)
+```
+
+# Downloading the reference sequences
+
+In order to perform the quantification of spliced and unspliced abundances with alevin [@srivastava2019alevin], we need to first prepare a suitable set of reference features (transcript isoform and intron sequences) to generate an index from. 
+We start by downloading the mouse reference genome and annotation gtf file from GENCODE.
+
+```{bash}
+wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M24/gencode.vM24.annotation.gtf.gz -P $workdir
+wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M24/GRCm38.primary_assembly.genome.fa.gz -P $workdir
+gunzip $workdir/GRCm38.primary_assembly.genome.fa.gz
+```
+
+# Extracting the transcript and intron sequences
+
+Next, we extract a GRanges object containing the genomic coordinates of each annotated transcript and intron. 
+
+```{r}
+gtf <- file.path(workdir, "gencode.vM24.annotation.gtf.gz")
+grl <- eisaR::getFeatureRanges(
+    gtf = gtf,
+    featureType = c("spliced", "intron"), 
+    intronType = "separate", 
+    flankLength = 90L, 
+    joinOverlappingIntrons = FALSE, 
+    verbose = TRUE
+)
+```
+
+After defining the genomic positions of all features of interest, we can extract the sequences of these, and write to a fasta file for later indexing with _Salmon_ [@patro2017salmon].
+In addition, we generate reference files that will be helpful for reading the data with _tximeta_ [@love2020tximeta] and splitting the joint count matrix into a spliced and an unspliced one.
+
+```{r}
+genome <- Biostrings::readDNAStringSet(
+    file.path(workdir, "GRCm38.primary_assembly.genome.fa")
+)
+names(genome) <- sapply(strsplit(names(genome), " "), .subset, 1)
+seqs <- GenomicFeatures::extractTranscriptSeqs(
+    x = genome, 
+    transcripts = grl
+)
+Biostrings::writeXStringSet(
+    seqs, filepath = file.path(workdir, "gencode.vM24.annotation.expanded.fa")
+)
+eisaR::exportToGtf(
+    grl, 
+    filepath = file.path(workdir, "gencode.vM24.annotation.expanded.gtf")
+)
+write.table(
+    metadata(grl)$corrgene, 
+    file = file.path(workdir, "gencode.vM24.annotation.expanded.features.tsv"),
+    row.names = FALSE, col.names = TRUE, quote = FALSE, sep = "\t"
+)
+df <- eisaR::getTx2Gene(
+    grl, filepath = file.path(workdir, "gencode.vM24.annotation.expanded.tx2gene.tsv")
+)
+```
+
+# Creating the quantification index
+
+We use _Salmon_ to generate a quantification index based on the expanded fasta file assembled above.
```

Comment:
Version of salmon and download location would be instructive.

Reply:
I've added the Salmon version and a link to the release